processing/src/main/java/org/apache/druid/frame/read/columnar/StringFrameColumnReader.java

```diff
@@ -507,6 +507,20 @@ protected Comparator<Object> getComparator()
       return Comparator.nullsFirst(Comparator.comparing(o -> ((String) o)));
     }
 
+    @Override
+    public int compareRows(int rowNum1, int rowNum2)
+    {
+      ByteBuffer buffer1 = getStringUtf8(rowNum1);
+      ByteBuffer buffer2 = getStringUtf8(rowNum2);
+
+      if (buffer1 == null) {
+        return -1;
```

Comment:
this is not necessarily true ; what if `buffer2 == null` ?
you could just use a commons-lang3 for this with
https://stackoverflow.com/a/10026065/1525291